01.Rmd

```diff
@@ -1,19 +1,199 @@
+text <- c("Because I could not stop for Death -",
+          "He kindly stopped for me -", 
+          "The Carriage held but just Ourselves -",
+          "and Immotality")
+text
+
+```
+
+We need to put this into a data frame to convert it into a tidy text dataset. 
+
+```{r 01-convert_dataframe, eval=TRUE}
+library(tidyverse)
+text_df <- tibble(line=1:4, text=text)
+text_df
+
+```
+
+Now, we can extract tokens (i.e., words in this example) from the data frame by using the `unnest_tokens` function.
+
+```{r 01-tokens, eval=TRUE}
+library(tidytext)
+text_df %>% 
+  unnest_tokens(word, text)
+```
+
+**unnest_tokens() function**
+- Other columns, such as the line number each word came from are retained. 
+- Punctuation has been stripped. 
+- By default, `unnest_tokens()` converts the tokens to lowercase, which makes them easier to compare or combine with other datasets (to_lower=FALSE to turn off this option)
+
+## Example 1: Tidying the works of Jane Austen
+
+**Loading the texts and converting it into a tibble data format:**
+
+```{r 01-example1, eval=TRUE}
+library(janeaustenr)
+library(dplyr)
+library(stringr)
+
+original_books <- austen_books() %>%
+  group_by(book) %>%
+  mutate(linenumber = row_number(),
+         chapter = cumsum(str_detect(text, 
+                                     regex("^chapter [\\divxlc]",
+                                           ignore_case = TRUE)))) %>%
+  ungroup()
+
+original_books
+```
+
+**Restructuring the one-token-per-row tidytext data format:**
+
+```{r 01-example1_unnest_token, eval=TRUE}
+library(tidytext)
+tidy_books <- original_books %>%
+  unnest_tokens(word, text)
+
+tidy_books
+```
+
+**Removing the stopwords:**
+Often in text analysis, we will want to remove stop words; stop words are words that are not useful for an analysis, typically extremely common words such as “the”, “of”, “to”, and so forth in English. We can remove stop words (kept in the tidytext dataset stop_words) with an `anti_join()`.
+
+```{r 01-example1_stopwords, eval=TRUE}
+data(stop_words)
+
+tidy_books <- tidy_books %>%
+  anti_join(stop_words)
+```
+
+We can use them all together, as we have here, or `filter()` to only use one set of stop words if that is more appropriate for a certain analysis.
+
+**Using the `dplyr::count()` to summarize the word frequency results as a tidy data table:**
+
+```{r 01-example1_countwords, eval=TRUE}
+tidy_books %>%
+  count(word, sort = TRUE) 
+```
+
+**Visualizing the word frequency results as a plot:**
+```{r 01-example1_ggplot, eval=TRUE}
+library(ggplot2)
+
+tidy_books %>%
+  count(word, sort = TRUE) %>%
+  filter(n > 600) %>%
+  mutate(word = reorder(word, n)) %>%
+  ggplot(aes(n, word)) +
+  geom_col() +
+  labs(y = NULL)
+```
+
+## Example 2: The `gutenbergr` package
+
+Check out the [ Problem with use of gutenberg_download function ](https://github.com/ropensci/gutenbergr/issues/8)
+
+```{r 01-example2_tidytexts, eval=TRUE}
+usethis::use_package("gutenbergr")
```

Comment:
No no! Run this in your local session, to add gutenbergr to the DESCRIPTION of *this* project. Do NOT call this inside the code.

Reply:
I saw your message and added gutenburgr and janeaustenr in the Imports part of the DESCRIPTION file. I hope this will be okay. Also,  I changed that usethis:: one into the previous one (library(gutenburgr)). That gutenburgr package is required for the Text Mining with R, but as a newbie for the GitHub, that package is killing me. 

My apologies for your confusion and inconvenience.

Reply:
No worries, this is a tricky thing to learn! Hopefully this will take care of it (and then we can actually review the content 😊 ).

janeaustenr should be fine, but adding it explicitly won't hurt!